File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.cpp
class Solution {
    void dfs(size_t i, size_t n, int target, std::vector<int>& candidates, std::vector<int>& sublist, std::vector<std::vector<int>>& res) {
        if (target == 0) {
            res.push_back(sublist);
            return;
        }
        if (i >= n || target < 0) {
            return;
        }
        sublist.push_back(candidates[i]);
        dfs(i+1, n, target - candidates[i], candidates, sublist, res);
        sublist.pop_back();
        while (i + 1 < n && candidates[i] == candidates[i + 1]) i++;
        dfs(i+1, n, target, candidates, sublist, res);
    }

public:
    vector<vector<int>> combinationSum2(vector<int>& candidates, int target) {
        std::sort(candidates.begin(), candidates.end());
        std::vector<int> sublist;
        std::vector<std::vector<int>> res;

        dfs(0, candidates.size(), target, candidates, sublist, res);
        return res;
    }
};

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-1.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        a, b, c = target
        i, j, k = False, False, False
        for x, y, z in triplets:
            # if (x == a and (y > b or z > c)) or (y == b and (x > a or z > c)) or (z == c and (x > a or y > b)):
            #     return False
            if x == a and y <= b and z <= c:
                i = True
            if y == b and x <= a and z <= c:
                j = True
            if z == c and x <= a and y <= b:
                k = True
        if not i or not j or not k:
            return False
        return True